leash.go

```diff
@@ -67,15 +74,36 @@ func run(options GlobalOptions) {
 		Options: options.Tail,
 	}
 	if options.TailSample {
-		linesChans, err = tail.GetSampledEntries(tc, options.SampleRate)
+		linesChans, err = tail.GetSampledEntries(tc, options.SampleRate, abort)
 	} else {
-		linesChans, err = tail.GetEntries(tc)
+		linesChans, err = tail.GetEntries(tc, abort)
 	}
 	if err != nil {
 		logrus.WithFields(logrus.Fields{"err": err}).Fatal(
 			"Error occurred while trying to tail logfile")
 	}
 
+	// set up our signal handler, now that we know how many files we're tailing,
+	// we can send the right number of abort signals.
+	go func() {
+		sig := <-sigs
+		fmt.Fprintf(os.Stderr, "Aborting! Caught signal \"%s\"\n", sig)
+		fmt.Fprintf(os.Stderr, "Cleaning up...\n")
+		close(abort)
+		// and if they insist, catch a second CTRL-C or timeout on 10sec
+		signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
https://husobee.github.io/golang/ecs/2016/05/19/ecs-graceful-go-shutdown.html

this is a cute pattern illustrating ^, in which their variant uses a channel to return the desired exit code (shrug, to each their own)